app/message_service.ts

```diff
@@ -31,5 +31,8 @@ export function getCount(tabId:number, callback: GenericCallback){
 export function clearData(tabId : number){
   chrome.runtime.sendMessage({message : "CLEAR_DATA", tabId})
 }
+export function interceptRequests(tabId:number,url:string, method:string, statusCode:number){
```

Comment:
It's a good practice to use an object for options if there are > 2 arguments. It can get confusing when removing the values. It's okay in TS, but goes south very quickly in JS.